mssql-ballerina/tests/connection-pool-test.bal

```diff
@@ -0,0 +1,586 @@
+//     resultArray[1] = dbClient1->query("select val = count(*)  from Customers where registrationID = 2", Result);
+
+//     resultArray[2] = dbClient2->query("select val = count(*)  from Customers where registrationID = 1", Result);
+//     resultArray[3] = dbClient2->query("select val = count(*)  from Customers where registrationID = 1", Result);
+
+//     resultArray[4] = dbClient3->query("select val = count(*)  from Customers where registrationID = 2", Result);
+//     resultArray[5] = dbClient3->query("select val = count(*)  from Customers where registrationID = 2", Result);
+
+//     resultArray[6] = dbClient4->query("select val = count(*)  from Customers where registrationID = 1", Result);
+//     resultArray[7] = dbClient4->query("select val = count(*)  from Customers where registrationID = 1", Result);
+
+//     resultArray[8] = dbClient5->query("select val = count(*)  from Customers where registrationID = 1", Result);
+//     resultArray[9] = dbClient5->query("select val = count(*)  from Customers where registrationID = 1", Result);
+
+//     resultArray[10] = dbClient5->query("select val = count(*)  from Customers where registrationID = 1", Result);
+
+//     (int|error)[] returnArray = [];
+//     int i = 0;
+//     // Connections will be released here as we fully consume the data in the following conversion function calls
+//     foreach var x in resultArray {
+//         returnArray[i] = getReturnValue(x);
+
+//         i += 1;
+//     }
+//     // All 5 clients are supposed to use the same pool. Default maximum no of connections is 10.
+//     // Since each select operation hold up one connection each, the last select operation should
+//     // return an error
+//     i = 0;
+//     while(i < 10) {
+//         test:assertEquals(returnArray[i], 1);
+//         i = i + 1;
+//     }
+//     validateConnectionTimeoutError(returnArray[10]);
+// }
+
+// isolated function getReturnValue(stream<record{}, error> queryResult) returns int|error {
+//     int count = -1;
+//     record {|record {} value;|}? data = check queryResult.next();
+//     if (data is record {|record {} value;|}) {
+//         record {} value = data.value;
+//         if (value is Result) {
+//             count = value.val;
+//         }
+//     }
+//     check queryResult.close();
+//     return count;
+// }
+
+// function validateApplicationError(int|error dbError) {
+//     test:assertTrue(dbError is error);
+//     sql:ApplicationError sqlError = <sql:ApplicationError> dbError;
+//     test:assertTrue(strings:includes(sqlError.message(), "Client is already closed"), sqlError.message());
+// }
+
+// function validateConnectionTimeoutError(int|error dbError) {
+//     test:assertTrue(dbError is error);
+//     sql:DatabaseError sqlError = <sql:DatabaseError> dbError;
+//     test:assertTrue(strings:includes(sqlError.message(), "request timed out after"), sqlError.message());
+// }
```

Comment:
Why these tests are commented out. shall we fix and enable them

Reply:
Fixed with [`b26f0c9`](https://github.com/ballerina-platform/module-ballerinax-mssql/pull/2/commits/b26f0c9ff3f06073099e4dbc658194f6dfb2d90f)